launcher/ui/pages/instance/ServersPage.cpp

```diff
@@ -112,11 +126,62 @@ struct Server {
     bool m_checked = false;
     bool m_up = false;
     QString m_motd;  // https://mctools.org/motd-creator
-    int m_ping = 0;
-    int m_currentPlayers = 0;
+    std::optional<int> m_currentPlayers; // nullopt if not calculated/calculating
     int m_maxPlayers = 0;
 };
 
+class ServerPingTask : public Task {
```

Comment:
consider moving the task to separate file

Reply:
Sure ! Where should that file be ? In `launcher/ui/pages/instance`, or somewhere else ?
I'd also love to move McResolver and McClient there, I feel like they don't belong in their current folder

Reply:
yeah in the `launcher/ui/pages/instance/` it should be on the same level as the other code added e.g. `McClient.cpp`